Top-level cargo test fails

It looks like top-level cargo test was introduced relatively recently, but I can't quite get it to work (tried on Windows with native Powershell and then Ubuntu in WSL, errors are the same):

```
running 3 tests
up to date in 1.702s

> neon-cli@0.1.20 transpile D:\Web\neon\cli
> tsc

src/cli.ts(258,9): error TS7030: Not all code paths return a value.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! neon-cli@0.1.20 transpile: `tsc`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the neon-cli@0.1.20 transpile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Ingvar\AppData\Roaming\npm-cache\_logs\2017-11-11T16_35_51_483Z-debug.log
test tests::cli_test ... FAILED
   Compiling neon-runtime v0.1.20 (file:///D:/Web/neon/crates/neon-runtime)
   Compiling neon v0.1.20 (file:///D:/Web/neon)
   Compiling static_tests v0.1.0 (file:///D:/Web/neon/test/static)
    Finished release [optimized] target(s) in 24.42 secs
     Running target\release\deps\run_all-fd0810a70eb331e7.exe

running 1 test

running 1 test
test [compile-fail] compile-fail\example_compile_error.rs ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

test run_all ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

test tests::static_test ... ok
up to date in 1.453s

> neon-cli@0.1.20 transpile D:\Web\neon\cli
> tsc

src/cli.ts(258,9): error TS7030: Not all code paths return a value.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! neon-cli@0.1.20 transpile: `tsc`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the neon-cli@0.1.20 transpile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Ingvar\AppData\Roaming\npm-cache\_logs\2017-11-11T16_36_31_842Z-debug.log
test tests::dynamic_test ... FAILED

failures:

---- tests::cli_test stdout ----
        thread 'tests::cli_test' panicked at 'assertion failed: Command::new(&shell).current_dir(dir).args(&[&command_flag,
                                             cmd]).status().expect("failed to execute test command").success()', src\lib.rs:306:8
note: Run with `RUST_BACKTRACE=1` for a backtrace.

---- tests::dynamic_test stdout ----
        thread 'tests::dynamic_test' panicked at 'assertion failed: Command::new(&shell).current_dir(dir).args(&[&command_flag,
                                             cmd]).status().expect("failed to execute test command").success()', src\lib.rs:306:8


failures:
    tests::cli_test
    tests::dynamic_test

test result: FAILED. 1 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--lib'
```

Looks like it all comes back from that single error `src/cli.ts(258,9): error TS7030: Not all code paths return a value.` which seems to relate to `return process.exit(1)` where, apparently, `process.exit(...)` returns `never` so TypeScript doesn't like it in context of async function.

Not sure why so, but splitting it into `process.exit(1); return;` seems to help and everything works after that.

--- a/cli/src/cli.ts
+++ b/cli/src/cli.ts
@@ -275,7 +275,8 @@ export default class CLI {
           break;
       }
 
-      return process.exit(1);
+      process.exit(1);
+      return;
     }
 
     try {

